eos-updater-prepare-volume/eos-updater-prepare-volume

```diff
@@ -194,13 +233,30 @@ class VolumePreparer:
         # TODO: Verify that it adds GPG keys where appropriate.
         _, repo = self.sysroot.get_repo()
         repo_path = os.path.realpath(repo.get_path().get_path())
-        self.__run(['ostree', 'summary', '--update', '--repo', repo_path])
+
+        # Increase the OSTree exclusive lock timeout significantly for the
+        # following OSTree commands.
+        #
+        # This reduces (but does not eliminate) the risk of another process
+        # holding the lock so long one of these commands gives up and this
+        # entire program fails.
+        self.ostree_lock_timeout_s = self._ostree_get_lock_timeout()
+        signal.signal(signal.SIGINT, self._sigint_handler)
+        self._ostree_set_lock_timeout(8*60)
+
         self.__run(['ostree', 'create-usb', '--repo', repo_path,
                     '--commit={}'.format(os_collection_ref[2]),
                     self.volume_path, os_collection_ref[0], os_collection_ref[1]])
+
         if len(all_flatpak_refs) > 0:
             self.__run(['flatpak', '--system', 'create-usb',
                         self.volume_path] + all_flatpak_refs)
+
+        # restore the original timeout so other OSTree processes wait the
+        # amount of time they normally would
+        self._ostree_restore_lock_timeout()
+        signal.signal(signal.SIGINT, signal.SIG_DFL)
```

Comment:
Instead of setting this to `signal.SIG_DFL`, you should save the earlier call to `signal.signal()` and reset it here. Python uses a different `SIGINT` handler to manage the `KeyboardInterrupt` exception.

Reply:
Fixed in the newly-pushed version